pynws/forecast.py

```diff
@@ -127,5 +127,12 @@ def get_details_by_hour(
             raise TypeError(f"{start_time!r} is not a datetime")
 
         start_time = start_time.replace(minute=0, second=0, microsecond=0)
-        for hour in range(hours):
-            yield self.get_details_for_time(start_time + timedelta(hours=hour))
+        for _ in range(hours):
+            end_time = start_time + ONE_HOUR
+            details = {
+                "startTime": datetime.isoformat(start_time),
```

Comment:
I think currently all custom info is in the `SimpleNWS` class.  One problem with this proposal is that `Detail` doesn't include these strings.  If we put it in the subclass then it could return according to an augmented Enum.

Thoughts?

Reply:
Good point about the Detail enum. The thing is that SimpleNWS doesn't currently do anything with DetailForecast. Thoughts?

Reply:
Sorry for the delay in responding. I think it makes sense where you have it.  We are already modifying the raw response here for easier digestion and the added information is very relevant to those modifications.

The only issue I see is that we cannot use all the attributes of `Detail` in the nonmodified details.  I don't have any good ideas here however.  I'm fine as is unless you have other thoughts.

Reply:
> We are already modifying the raw response here for easier digestion and the added information is very relevant to those modifications.

> The only issue I see is that we cannot use all the attributes of `Detail` in the nonmodified details. I don't have any good ideas here however. I'm fine as is unless you have other thoughts.

The detail format is a new format created from elements of the raw format. Adding the start and end times to the detail format is quite useful for responses from the get_details_by_hour method. I'm happy with it the way it is now.